pkg/cloudprovider/instance.go

```diff
@@ -169,6 +169,7 @@ func (p *InstanceProvider) launchInstance(ctx context.Context, provider *v1alpha
 	if err != nil {
 		if awserrors.IsLaunchTemplateNotFound(err) {
 			for _, lt := range launchTemplateConfigs {
+				ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("launch-template-name", aws.StringValue(lt.LaunchTemplateSpecification.LaunchTemplateName), "launch-template-id", aws.StringValue(lt.LaunchTemplateSpecification.LaunchTemplateId)))
```

Comment:
This continually mutates the ctx variable, making an ancestry of children for each loop. One way to avoid this is by making a new variable `innerCtx := ...`. Another way is to just do it inside the Invalidate call. I prefer the latter.